### PR TITLE
feat: filter by PURL type

### PR DESCRIPTION
**Pull request by dervoeti**
This change adds the ability to filter observations by the PURL type of the affected component. I found that helpful when trying to focus on, for example, only the RPM packages after scanning a container image.

### CODE REVIEW

File: backend/application/core/models.py [hunk 1]
```diff
@@ -520,6 +520,7 @@ class Observation(Model):
         on_delete=SET_NULL,
     )
     risk_acceptance_expiry_date = DateField(null=True)
+    purl_type = CharField(max_length=16, blank=True, null=True)
```
Comment (StefanFl) [inline backend/application/core/models.py:523]: This attribute shall have the name `origin_component_purl_type` and shall be placed directly behind `origin_component_purl`.

This name change shall be reflected in the other files using this attribute as well.

File: backend/application/import_observations/parsers/cyclone_dx/parser.py [hunk 2]
```diff
@@ -111,6 +112,11 @@ def _create_observations(  # pylint: disable=too-many-locals
                             data, components, component, metadata
                         )
 
+                        purl_type = None
```
Comment (StefanFl) [inline backend/application/import_observations/parsers/cyclone_dx/parser.py:115]: This section shall be placed in method `normalize_origin_component` of `application.core.services.observation.py`. Then it is executed always when there are changes in an observation, no matter where the change occurs.

**Review by StefanFl — CHANGES_REQUESTED** [review]

File: backend/application/core/models.py (hunk 1 above)
Comment (dervoeti) [reply backend/application/core/models.py]: Done

File: backend/application/import_observations/parsers/cyclone_dx/parser.py (hunk 2 above)
Comment (dervoeti) [reply backend/application/import_observations/parsers/cyclone_dx/parser.py]: Done

File: backend/unittests/access_control/api/test_authorization_observation_logs.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def _test_authorization_observation_logs(self):
...
...ast_seen': '2022-12-15T17:14:20.870000+01:00', 'created': '2022-12-15T17:10:35.513000+01:00', 'modified': '2022-12-16T17:13:18.282000+01:00', 'last_observation_log': '2022-12-16T17:13:18.281000+01:00', 'identity_hash': '6eef8088480aa2523aeeb64ad35f876a942cc3172cfb36752f3a052a4f88642c', 'issue_tracker_issue_id': '', 'issue_tracker_issue_closed': False, 'has_potential_duplicates': False, 'current_vex_justification': '', 'parser_vex_justification': '', 'vex_vex_justification': '', 'rule_vex_justification': '', 'assessment_vex_justification': '', 'risk_acceptance_expiry_date': None, 'origin_component_purl_type': None, 'product': 1, 'branch': 1, 'parser': 1, 'origin_service': 1, 'general_rule': None, 'product_rule': 1, 'vex_statement': None}, 'user_full_name': 'db_internal_write', 'approval_user_full_name': None, 'severity': 'Medium', 'status': 'Open', 'comment': 'Set by parser', 'created': '2022-12-15T17:10:35.518000+01:00', 'vex_justification': '', 'assessment_status': 'Auto approved', 'approval_remark': '', 'approval_date': None, 'risk_acceptance_expiry_date': None, 'observation': 1, 'user': 2, 'approval_user': None, 'general_rule': None, 'product_rule': None, 'vex_statement': None}"
```
Comment (StefanFl) [inline backend/unittests/access_control/api/test_authorization_observation_logs.py:38]: Please fix this and the other unittests, then we are ready to merge.

Comment (dervoeti) [reply backend/unittests/access_control/api/test_authorization_observation_logs.py]: @StefanFl Thanks for sending over the fixed unittests, I commited them.

**Review by StefanFl — APPROVED** [review]

**Pull request merged**